pkg/agent/core/ngt/service/kvs/kvs_test.go

```diff
@@ -507,65 +757,147 @@ func Test_bidi_DeleteInverse(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           val: 0,
-		       },
-		       fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           val: 0,
-		           },
-		           fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+			var wantl = 0
+
+			return test{
+				name: "return (45637ec4-c85f-11ea-87d0, true) when the delete successes",
+				args: args{
+					val: val,
+				},
+				beforeFunc: func(a args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				checkFunc: func(w want, a args, b *bidi, gotKey string, gotOk bool) error {
+					if err := defaultCheckFunc(w, a, b, gotKey, gotOk); err != nil {
+						return err
+					}
+					if l := atomic.LoadUint64(&b.l); wantl != 0 {
+						return errors.Errorf("l is not equals.\twant: %v, but got: %v", wantl, l)
+					}
+					if _, ok := b.Get(key); ok {
+						return errors.New("the value for the key exists")
+					}
+					if _, ok := b.GetInverse(val); ok {
+						return errors.New("the key for the val has not disappeared")
+					}
+					return nil
+				},
+				fields: fields,
+				want: want{
+					wantKey: key,
+					wantOk:  true,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key = "45637ec4-c85f-11ea-87d0"
+			)
+			var wantl = 0
+
+			return test{
+				name: "return (45637ec4-c85f-11ea-87d0, true) when the delete successes and the val is default value",
```

Comment:
same

Reply:
thank you
I fixed: https://github.com/vdaas/vald/pull/1223/commits/06e4a0ddeaaf71d1c1958a078f3de4e392eb9fb4